index.bs

```diff
@@ -3212,6 +3215,69 @@ To <dfn>delete expired sources</dfn> given a [=moment=] |now|:
     1. If |source|'s [=attribution source/expiry time=] is less than |now|,
         [=set/remove=] |source| from the [=attribution source cache=].
 
+To <dfn>find sources with common destinations and reporting origin</dfn> given an [=attribution source=] |pendingSource|:
+1. Let |matchingSources| be a new [=list=].
+1. [=set/iterate|For each=] |source| of the user agent's [=attribution source cache=]:
+    1. Let |commonDestinations| be the [=set/intersection=] of |source|'s [=attribution source/attribution destinations=] and |pendingSource|'s [=attribution source/attribution destinations=].
+    1. If |commonDestinations| [=set/is empty=], [=iteration/continue=].
+    1. If |source|'s [=attribution source/reporting origin=] and |pendingSource|'s [=attribution source/reporting origin=] are not [=same origin=], [=iteration/continue=].
+    1. [=list/Append=] |source| to |matchingSources|.
+1. Return |matchingSources|.
+
+To <dfn>check if source has incompatible attribution scope fields</dfn> given an [=attribution source=] |source| and an [=attribution source=] |pendingSource|:
+1. If |pendingSource|'s [=attribution source/attribution scope limit=] is null:
+    1. [=Assert=]: |source|' [=attribution source/attribution scope limit=] is null.
```

Comment:
The `Assert` here seems a little confusing. Wdyt about reorganizing the check as the follows:

1.  In `remove or update sources with incompatible attribution scope fields`:
```
    1. If the result of running [=check if source has incompatible attribution scope fields=] with |source| and |pendingSource| is false, [iteration/continue=].
    1. If |pendingSource|'s [=attribution source/attribution scope limit=] is null and |source|'s [=attribution source/attribution scope limit=] is not null:
        1. Set |source|'s [=attribution source/attribution scope limit=] to null.
        1. Set |source|'s [=attribution source/attribution scopes=] to a new [=set=].
        1. Set |source|'s [=attribution source/max event states=] to [=default max event states=].
    1. Otherwise:
        1. [=Remove associated event reports and rate-limit records=] with |source|'s [=attribution source/source identifier=] and |pendingSource|'s [=attribution source/source time=].
        1. [=set/Remove=] |source| from the [=attribution source cache=].
```
2. Return  whether `|source|'s [=attribution source/attribution scope limit=] is null` in this algorithm.

Reply:
Good point. Done.